src/server/json_family.cc

```diff
@@ -1822,6 +1828,11 @@ void JsonFamily::StrLen(CmdArgList args, ConnectionContext* cntx) {
   Transaction* trans = cntx->transaction;
   auto result = trans->ScheduleSingleHopT(std::move(cb));
   auto* rb = static_cast<RedisReplyBuilder*>(cntx->reply_builder());
+  if (result == OpStatus::KEY_NOTFOUND) {
+    if (path.empty())
```

Comment:
We can use `if (json_path.IsLegacyModePath())` here. Thus, null will be returned for the both cases `json.strlen json2` and `json.strlen json2 .a`.

Because we are parsing an empty path as a path in legacy mode.